src/interface/meshblock_data.hpp

```diff
@@ -461,8 +431,7 @@ class MeshBlockData {
   MapToVariablePack<T> coarseVarPackMap_; // cache for varpacks over coarse arrays
   MapToVariableFluxPack<T> varFluxPackMap_;
 
-  void calcArrDims_(std::array<int, 6> &arrDims, const std::vector<int> &dims,
-                    const Metadata &metadata);
+  std::array<int, 6> calcArrDims_(const Metadata &metadata);
```

Comment:
Are there any performance implications of this change? or does the move operator save us?

Reply:
Good question. The compiler should do a return value optimization here, meaning it shouldn't make an unnecessary copy of the `std::array`, so I think there shouldn't be any performance penalty. Maybe it would even perform slightly better because we save a pointer lookup :)